### ISSUE TITLE
Changefeed "Unreachable code" crash on null field

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#5535]
This one looks similar to https://github.com/rethinkdb/rethinkdb/issues/5438 , but still happens on `v2.2.x`:

```js
r.tableCreate('t1')
r.table('t1').insert({v: null})
r.table('t1').getField('v').changes({includeInitial: true})
```

```c++
error: Error in src/rdb_protocol/datum.cc at line 1475:
error: Unreachable code: 
error: Backtrace:
error: Thu Mar 17 13:32:06 2016
       
       1 [0xc928dc]: backtrace_t::backtrace_t() at backtrace.cc:204 (discriminator 1)
       2 [0xc94263]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:325
       3 [0xc922d4]: format_backtrace(bool) at backtrace.cc:199
       4 [0xc94f5d]: report_fatal_error(char const*, int, char const*, ...) at basic_string.h:293
       5 [0xbd570a]: void ql::datum_t::write_json_unchecked_stack<rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::RAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::RAllocator> >(rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::RAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::RAllocator>*) const at datum.cc:1475
       6 [0xbd22b6]: void ql::datum_t::write_json<rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::RAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::RAllocator> >(rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::RAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::RAllocator>*) const at datum.cc:1484
       7 [0xc66d3b]: write_response_internal(ql::response_t*, rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::RAllocator>*, bool) at stl_iterator.h:748
       8 [0xc6b74c]: query_server_t::handle(http_req_t const&, http_res_t*, signal_t*) at stack.h:133
       9 [0xa16ef1]: routing_http_app_t::handle(http_req_t const&, http_res_t*, signal_t*) at routing_app.cc:29 (discriminator 2)
       10 [0xa16ef1]: routing_http_app_t::handle(http_req_t const&, http_res_t*, signal_t*) at routing_app.cc:29 (discriminator 2)
       11 [0xa1e9bd]: http_server_t::handle_conn(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t) at http.cc:413
       12 [0xa21744]: void boost::_bi::list3<boost::_bi::value<http_server_t*>, boost::arg<1>, boost::_bi::value<auto_drainer_t::lock_t> >::operator()<boost::_mfi::mf2<void, http_server_t, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t>, boost::_bi::list1<scoped_ptr_t<linux_tcp_conn_descriptor_t>&> >(boost::_bi::type<void>, boost::_mfi::mf2<void, http_server_t, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t>&, boost::_bi::list1<scoped_ptr_t<linux_tcp_conn_descriptor_t>&>&, int) at mem_fn_template.hpp:280 (discriminator 4)
       13 [0xa2162a]: std::_Function_handler<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&), boost::_bi::bind_t<void, boost::_mfi::mf2<void, http_server_t, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t>, boost::_bi::list3<boost::_bi::value<http_server_t*>, boost::arg<1>, boost::_bi::value<auto_drainer_t::lock_t> > > >::_M_invoke(std::_Any_data const&, scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at functional:2073
       14 [0x720870]: linux_nonthrowing_tcp_listener_t::handle(int) at scoped.hpp:70
       15 [0x7262e8]: coro_t::run() at coroutines.cc:214
error: Exiting.
Trace/breakpoint trap (core dumped)
```

Assigning @mlucy

**Comment by mlucy** [comment]
A fix for this is in CR 3533 by @danielmewes .

**Comment by mlucy** [comment]
The fix is in next and 2.2.x.
